File: 0701_0800/0798/0798.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    n, m, i, j, c = (int(x) for x in f.readline().split())
    q.write("equal" if not (m % 2 and n % 2) else "white" if ((i - 1) * n + j) % 2 == c else "black")

File: 0701_0800/0798/0798.c
#include <stdio.h>

int main()
{
	FILE *f = fopen("input.txt", "r"), *q = fopen("output.txt", "w");
	long long n, m, i, j, c;

	fscanf(f, "%I64d%I64d%I64d%I64d%I64d", &n, &m, &i, &j, &c);
	fprintf(q, !(m % 2 && n % 2) ? "equal" : (((i - 1) * n + j) % 2 == c) ? "white" : "black");
	return 0;
}
